guacamole/src/main/frontend/src/app/client/controllers/clientController.js

```diff
@@ -468,6 +483,31 @@ angular.module('client').controller('clientController', ['$scope', '$routeParams
         else if (menuShownPreviousState)
             $scope.applyParameterChanges($scope.focusedClient);
 
+        /* Broadcast changes to the menu display state */
+        $scope.$broadcast('guacMenuShown', menuShown);
+
+    });
+
+    // Toggle the menu when the guacClientToggleMenu event is received
+    $scope.$on('guacToggleMenu',
+            () => $scope.menu.shown = !$scope.menu.shown);
+
+    // Show the menu when the guacClientShowMenu event is received
+    $scope.$on('guacShowMenu', () => $scope.menu.shown = true);
+
+    // Hide the menu when the guacClientHideMenu event is received
+    $scope.$on('guacHideMenu', () => $scope.menu.shown = false);
+
+    // Broadcast any mouse events caught from clients back down to the rest
+    // of the client page
+    $scope.$on('guacClientMouseEvent', (angularEvent, mouseEvent) => {
+        $scope.$broadcast('guacMouseEvent', mouseEvent);
+    });
+
+    // Broadcast any touch events caught from clients back down to the rest
+    // of the client page
+    $scope.$on('guacClientTouchEvent', (angularEvent, touchEvent) => {
+        $scope.$broadcast('guacTouchEvent', touchEvent);
```

Comment:
The client versions bubble up from the client directive to the page controller, which then broadcasts the non-client versions down to all its children. This would allow a field in the client menu to hook into these to implement e.g. drag and drop functionality, which would otherwise be precluded because the client directive eats up all the mouse and touch events.

Reply:
Can that handling not be left up to whatever code lies behind that custom field? If there's a need for the field to intercept the events that would normally reach the client, I think that can be done by briefly adding an overlay element in front of the client view.

Reply:
That sounds like a much better idea - I'll give it a try.

Reply:
Yep, it works. Removing these events from this PR.